controllers/loadtest_controller.go

```diff
@@ -139,12 +155,38 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	}
 
+	reQueueTime := getRequeueTime(test, previousStatus, log)
+	if reQueueTime != 0 {
+		return ctrl.Result{RequeueAfter: reQueueTime}, nil
+	}
 	return ctrl.Result{}, nil
 }
 
-// LoadTestMissing categorize missing components based on their roles at specific
-// moment. The struct is a wrapper to help us get role information associate
-// with components.
+// getRequeueTime takes a LoadTest and its previous state, and compares the
+// previous status of the load test with its updated status and return a
+// calculated requeue time. The reason we take the test itself other than
+// its current status is because the ttl and time out are saved in its spec.
+// if the load test has just been assigned a start time, getRequeueTime returns
+// the timeout specified within its spec. If the load test has been just
+// assigned stop time, getRequeueTime returns its ttl specified within its spec
+// less its actual running time. In other cases, returns a 0 value time.duration
+func getRequeueTime(updatedLoadTest *grpcv1.LoadTest, previousStatus grpcv1.LoadTestStatus, log logr.Logger) time.Duration {
+	reQueueTime := time.Duration(0)
+
+	if previousStatus.StartTime == nil && updatedLoadTest.Status.StartTime != nil {
```

Comment:
What would happen in the following case? The controller sets the start time and requeues, but then the controller crashes and restarts. The queue is lost. How would this logic be triggered again? I have no idea how we'd handle that case.

Reply:
I feel there is not any convenient way to deal with re queue of the event.

The worst case scenario:
1. Not re queue with timeout --> not able to catch it when it did went over time -->  the cleanup agent won't know about this --> would not stop the workers if they are still running --> will depend on TTL to delete them.
This worse case scenario would cause some resource waste, but if it happened too frequently, user interference might be required to resolved it. 

2. Not re queue to see it at TTL --> the load test will live on the cluster possibly forever --> possibility causing resource waste, this again would require human attention if this happened a lot. 

I agree it would be nice to have something that independent,  just do a cleanup up for everything on the cluster (delete the ones live out of TTL), I also agree this is more like a improvement later, if we do need this, it should not be too difficult to add down the road.